java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/core/recordretrieval/BulkExportQuerySplitter.java

```diff
@@ -0,0 +1,224 @@
+     *
+     * @param partitions             to used to initialise the class
+     * @param partitionToFileMapping maps the partitions to files
+     */
+    public void init(List<Partition> partitions, Map<String, List<String>> partitionToFileMapping) {
+        init(partitions, partitionToFileMapping, Instant.now());
+    }
+
+    /**
+     * Initialises the partitions and the mapping from partitions to active files if
+     * needed.
+     *
+     * @param now used to determine the next initialisation.
+     * @throws StateStoreException if the statestore can't be accessed.
+     */
+    public void initIfNeeded(Instant now) throws StateStoreException {
+        if (nextInitialiseTime.isAfter(now)) {
+            LOGGER.debug("Not refreshing state for table {}", tableProperties.getStatus());
+            return;
+        }
+        init(now);
+    }
+
+    /**
+     * Initialises the partitions and the mapping from partitions to active files.
+     *
+     * @param now used to determine the next initialisation.
+     * @throws StateStoreException if the statestore can't be accessed.
+     */
+    public void init(Instant now) throws StateStoreException {
+        List<Partition> partitions = stateStore.getAllPartitions();
+        Map<String, List<String>> partitionToFileMapping = stateStore.getPartitionToReferencedFilesMap();
+        init(partitions, partitionToFileMapping, now);
+    }
+
+    /**
+     * Initialises the partitions and the mapping from partitions to active files.
+     *
+     * @param partitions             to used to initialise the class
+     * @param partitionToFileMapping maps the partitions to files
+     * @param now                    used to determine the next initialisation.
+     */
+    public void init(List<Partition> partitions, Map<String, List<String>> partitionToFileMapping, Instant now) {
+        leafPartitions = partitions.stream()
+                .filter(Partition::isLeafPartition)
+                .collect(Collectors.toList());
+        partitionTree = new PartitionTree(partitions);
+        partitionToFiles = partitionToFileMapping;
+        nextInitialiseTime = now.plus(tableProperties.getInt(QUERY_PROCESSOR_CACHE_TIMEOUT), ChronoUnit.MINUTES);
+        LOGGER.info("Loaded state for table {}. Found {} partitions. Next initialise time: {}",
+                tableProperties.getStatus(), partitions.size(), nextInitialiseTime);
+    }
+
+    /**
+     * Executes a bulk export query. This method first splits up the query into one
+     * or more BulkExportLeafPartitionQuery.
+     *
+     * @param bulkExportQuery the bulk export query
+     * @return a list of LeafPartitionQuery
+     */
+    public List<BulkExportLeafPartitionQuery> execute(BulkExportQuery bulkExportQuery) {
+        List<BulkExportLeafPartitionQuery> leafPartitionQueries = splitIntoLeafPartitionQueries(bulkExportQuery);
+        return leafPartitionQueries;
```

Comment:
It looks like this method is just a duplicate of splitIntoLeafPartitionQueries? Can we remove it?

Reply:
Removed method